[BUG] Jar hell from org.bouncycastle:bcprov-jdk18on

### What is the bug?

Build is failing on Jar Hell.
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':dependencies'.
> Could not resolve all dependencies for configuration ':testCompileClasspath'.
   > Conflict found for the following module:
       - org.bouncycastle:bcprov-jdk18on between versions 1.80 and 1.78
```

### How can one reproduce the bug?

 -  `./gradlew compileTestJava`

### What is the expected behavior?

No Jar Hell

### Do you have any additional context?

Our build.gradle specifies version 1.8.0:
https://github.com/opensearch-project/flow-framework/blob/88fe98e561a68337fbe1188e0189a87a01fa4f50/build.gradle#L180

The OpenSearch Version Catalog specifies 1.7.8:
https://github.com/opensearch-project/OpenSearch/blob/0ffed5e8b743a075f5d66c4fe6e9b8371eacfa14/gradle/libs.versions.toml#L61

Quick fix:
 - Change the `build.gradle` line above to get the version from the version catalog (`${versions.bouncycastle}`)

Better fix:
 - Audit all the dependencies to see if they are in the version catalog and update appropriately.
 - Check whether they are actually required on the main build or whether they can be `testImplementation` as this one appears to probably be

--- a/build.gradle
+++ b/build.gradle
@@ -177,7 +177,7 @@ dependencies {
     implementation "software.amazon.cryptography:aws-cryptographic-material-providers:1.9.0"
     implementation "org.dafny:DafnyRuntime:4.10.0"
     implementation "software.amazon.smithy.dafny:conversion:0.1.1"
-    implementation 'org.bouncycastle:bcprov-jdk18on:1.80'
+    implementation "org.bouncycastle:bcprov-jdk18on:${versions.bouncycastle}"
     api "org.apache.httpcomponents.core5:httpcore5:${versions.httpcore5}"
     implementation "jakarta.json.bind:jakarta.json.bind-api:3.0.1"
     implementation "org.glassfish:jakarta.json:2.0.1"

